nft/src/abi.sw

```diff
@@ -0,0 +1,32 @@
+library abi;
+
+use std::{identity::Identity, option::Option};
+
+abi NFT {
+    #[storage(read, write)]
+    fn allow_mint(minter: Identity, allow: bool);

```

Comment:
I feel that this function is out of place in the abi, and would prefer a simplified (but still flexible) approach.
e.g: Set a single `minter: Identity` in the constructor.

This minter can be an EOA for simple scenarios, or can be a dedicated `Minter` contract which can modulate access (using its own access control scheme to add/remove Identities which can mint NFT _through_ the `Minter` contract.
This would allow for arbitrarily complex access control schemes around minting, while keeping the base NFT contract clean and minimal.

Reply:
The dedicated `Minter` contract could also be upgraded while retaining the same ContractId, allowing yet another path for making changes without complicating the base NFT.

Reply:
I like this approach a lot. I will likely create an example `Minter` contract in the near future